djangocms_bootstrap4/contrib/bootstrap4_grid/cms_plugins.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from djangocms_bootstrap4.constants import DEVICE_SIZES
+from djangocms_bootstrap4.helpers import concat_classes
+
+from .models import (
+    Bootstrap4GridContainer,
+    Bootstrap4GridRow,
+    Bootstrap4GridColumn,
+)
+from .forms import (
+    Bootstrap4GridRowForm,
+    Bootstrap4GridColumnForm,
+)
+
+
+class Bootstrap4GridContainerPlugin(CMSPluginBase):
+    """
+    Layout > Grid: "Container" Plugin
+    https://getbootstrap.com/docs/4.0/layout/grid/
+    """
+    model = Bootstrap4GridContainer
+    name = _('Container')
+    module = _('Bootstrap 4')
+    render_template = 'djangocms_bootstrap4/grid_container.html'
+    allow_children = True
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'container_type',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'tag_type',
+                'attributes',
+            )
+        }),
+    ]
+
+    def render(self, context, instance, placeholder):
+        classes = concat_classes([
+            instance.container_type,
+            instance.attributes.get('class'),
+        ])
+        instance.attributes['class'] = classes
```

Comment:
context variables please

Reply:
just for further reference, this was done so that attributes in the template does not render ``class`` twice, once for attributes, once for context